src/mds/Migrator.cc

```diff
@@ -947,23 +1106,91 @@ void Migrator::dispatch_export_dir(MDRequestRef& mdr, int count)
   }
 
   assert(g_conf()->mds_kill_export_at != 1);
-  it->second.state = EXPORT_DISCOVERING;
-
-  // send ExportDirDiscover (ask target)
-  filepath path;
-  dir->inode->make_path(path);
-  MExportDirDiscover *discover = new MExportDirDiscover(dir->dirfrag(), path,
-							mds->get_nodeid(),
-							it->second.tid);
-  mds->send_message_mds(discover, dest);
-  assert(g_conf()->mds_kill_export_at != 2);
-
-  it->second.last_cum_auth_pins_change = ceph_clock_now();
-
-  // start the freeze, but hold it up with an auth_pin.
-  dir->freeze_tree();
-  assert(dir->is_freezing_tree());
-  dir->add_waiter(CDir::WAIT_FROZEN, new C_MDC_ExportFreeze(this, dir, it->second.tid));
+
+  vector<pair<CDir*, size_t> > results;
+  maybe_split_export(dir, results);
+
+  auto now = ceph_clock_now();
+  if (results.size() == 1 && results.front().first == dir) {
```

Comment:
What circumstance would `results.front().first == dir` not be true if `results.size() == 1`?

Reply:
original subtree A
[pre]
   A
/    \
B    C
[/pre]

maybe_split_export only wants to export subtree B (subtree B is large enough)